Locally, I do not get the malformed cell.scala and TcpTransportLayerSpec.scala errors. PLEASE RESOLVE. Thanks

When I ran with catsEffect 2.1.4 I got the following errors:
@nzpr @zsluedem @tgrospic The main failure on all the failed unit tests seems to be the following (not sure if these are related to the required changes to TwoStepLockTest.scala). Could one of you please fix the two errors and take it forward? Thanks much..
[error] /__w/rchain/rchain/casper/src/test/scala/coop/rchain/casper/helper/BlockDagStorageFixture.scala:55:17: wrong number of type parameters for method use: [G[x] >: monix.eval.Task[x], B](f: ((coop.rchain.blockstorage.BlockStore[monix.eval.Task], coop.rchain.blockstorage.dag.IndexedBlockDagStorage[monix.eval.Task], coop.rchain.casper.util.rholang.RuntimeManager[monix.eval.Task])) => G[B])(implicit F: cats.effect.Bracket[G,Throwable])G[B]
[error]     resource.use[R] { case (b, d, r) => f(b)(d)(r) }.unsafeRunSync
[error]                 ^
[error] /__w/rchain/rchain/casper/src/test/scala/coop/rchain/casper/helper/BlockDagStorageFixture.scala:55:43: type mismatch;
[error]  found   : Any
[error]  required: coop.rchain.blockstorage.BlockStore[monix.eval.Task]
[error]     resource.use[R] { case (b, d, r) => f(b)(d)(r) }.unsafeRunSync
[error]                                           ^
[error] two errors found
[error] (casper / Test / compileIncremental) Compilation failed
[error] Total time: 166 s, completed Oct 12, 2020, 8:25:08 PM
Error: Process completed with exit code 1.

--- a/casper/src/main/scala/coop/rchain/casper/ReportingCasper.scala
+++ b/casper/src/main/scala/coop/rchain/casper/ReportingCasper.scala
@@ -1,7 +1,7 @@
 package coop.rchain.casper
 
 import cats.data.EitherT
-import cats.effect.concurrent.{MVar, Ref}
+import cats.effect.concurrent.{MVar, MVar2, Ref}
 import cats.effect.{Concurrent, ContextShift, Sync}
 import cats.implicits._
 import cats.{Monad, Parallel}
@@ -260,7 +260,7 @@ object ReportingCasper {
     */
   class ReportingRuntimeManagerImpl[F[_]: Concurrent: Metrics: Span: Log](
       val emptyStateHash: StateHash,
-      runtimeContainer: MVar[F, ReportingRuntime[F]]
+      runtimeContainer: MVar2[F, ReportingRuntime[F]]
   ) {
     import coop.rchain.models.rholang.{implicits => toPar}
     private val systemDeployConsumeAllPattern =

--- a/casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala
+++ b/casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala
@@ -2,7 +2,7 @@ package coop.rchain.casper.util.rholang
 
 import cats.Applicative
 import cats.data.{EitherT, WriterT}
-import cats.effect.concurrent.MVar
+import cats.effect.concurrent.{MVar, MVar2}
 import cats.effect.{Sync, _}
 import cats.syntax.all._
 import com.google.protobuf.ByteString
@@ -83,7 +83,7 @@ trait RuntimeManager[F[_]] {
 
 class RuntimeManagerImpl[F[_]: Concurrent: Metrics: Span: Log](
     val emptyStateHash: StateHash,
-    runtimeContainer: MVar[F, Runtime[F]]
+    runtimeContainer: MVar2[F, Runtime[F]]
 ) extends RuntimeManager[F] {
   import coop.rchain.models.rholang.{implicits => toPar}
 

--- a/project/Dependencies.scala
+++ b/project/Dependencies.scala
@@ -8,9 +8,9 @@ object Dependencies {
   val enumeratumVersion = "1.5.13"
   val http4sVersion     = "0.21.4"
   val kamonVersion      = "1.1.5"
-  val catsVersion       = "2.1.0"
-  val catsEffectVersion = "2.0.0"
-  val catsMtlVersion    = "0.7.0"
+  val catsVersion       = "2.1.1"
+  val catsEffectVersion = "2.2.0"
+  val catsMtlVersion    = "0.7.1"
   val slf4jVersion      = "1.7.25"
 
   // format: off
@@ -24,7 +24,7 @@ object Dependencies {
   val catsMtl             = "org.typelevel"              %% "cats-mtl-core"             % catsMtlVersion
   val catsMtlLawsTest     = "org.typelevel"              %% "cats-mtl-laws"             % catsMtlVersion % "test"
   val catsTagless         = "org.typelevel"              %% "cats-tagless-macros"       % "0.11"
-  val disciplineCore      = "org.typelevel"                                    %% "discipline-core"           % "1.0.2"
+  val disciplineCore      = "org.typelevel"              %% "discipline-core"           % "1.0.3"
   val circeCore           = "io.circe"                   %% "circe-core"                % circeVersion
   val circeGeneric        = "io.circe"                   %% "circe-generic"             % circeVersion
   val circeGenericExtras  = "io.circe"                   %% "circe-generic-extras"      % circeVersion

--- a/project/Dependencies.scala
+++ b/project/Dependencies.scala
@@ -52,7 +52,7 @@ object Dependencies {
   val lmdbjava            = "org.lmdbjava"                % "lmdbjava"                  % "0.8.1"
   val logbackClassic      = "ch.qos.logback"              % "logback-classic"           % "1.2.3"
   val lz4                 = "org.lz4"                     % "lz4-java"                  % "1.5.1"
-  val monix               = "io.monix"                   %% "monix"                     % "3.1.0"
+  val monix               = "io.monix"                   %% "monix"                     % "3.2.2"
   val scalaLogging        = "com.typesafe.scala-logging" %% "scala-logging"             % "3.9.2"
   val scalaUri            = "io.lemonlabs"               %% "scala-uri"                 % "1.1.5"
   val scalacheck          = "org.scalacheck"             %% "scalacheck"                % "1.14.3"

--- a/shared/src/main/scala/coop/rchain/shared/Cell.scala
+++ b/shared/src/main/scala/coop/rchain/shared/Cell.scala
@@ -2,7 +2,7 @@ package coop.rchain.shared
 
 import cats._
 import cats.data.ReaderT
-import cats.effect.concurrent.{MVar, Ref}
+import cats.effect.concurrent.{MVar, MVar2, Ref}
 import cats.effect.{Concurrent, ExitCase, Sync}
 import cats.syntax.flatMap._
 import cats.syntax.functor._
@@ -38,7 +38,8 @@ object Cell {
     override def reads[A](f: S => A) = read map f
   }
 
-  private class MVarCell[F[_], S](mvar: MVar[F, S])(implicit F: Sync[F]) extends DefaultCell[F, S] {
+  private class MVarCell[F[_], S](mvar: MVar2[F, S])(implicit F: Sync[F])
+      extends DefaultCell[F, S] {
     def modify(f: S => S): F[Unit] = flatModify(s => F.pure(f(s)))
     def flatModify(f: S => F[S]): F[Unit] = F.bracketCase(mvar.take)(s => f(s).flatMap(mvar.put)) {
       case (oldState, ExitCase.Error(_) | ExitCase.Canceled) => mvar.put(oldState)

